File: domain/src/main/java/com/beat/domain/staff/repository/StaffRepository.java
package com.beat.domain.staff.repository;

import java.util.List;
import java.util.Optional;

import com.beat.domain.staff.domain.Staff;

public interface StaffRepository {

	Optional<Staff> findById(Long staffId);

	List<Staff> findByPerformanceId(Long performanceId);

	List<Staff> findAllByPerformanceId(Long performanceId);

	List<Long> findIdsByPerformanceId(Long performanceId);

	Staff save(Staff staff);

	List<Staff> saveAll(List<Staff> staffs);

	void delete(Staff staff);

	void deleteByPerformanceId(Long performanceId);
}

File: domain/src/main/kotlin/com/beat/domain/staff/repository/StaffRepository.kt
package com.beat.domain.staff.repository

import com.beat.domain.staff.domain.Staff
import java.util.*

@JvmSuppressWildcards
interface StaffRepository {
    fun findById(staffId: Long?): Optional<Staff>

    fun findAllByPerformanceId(performanceId: Long?): List<Staff>

    fun findIdsByPerformanceId(performanceId: Long?): List<Long>

    fun save(staff: Staff): Staff

    fun saveAll(staffs: List<Staff>): List<Staff>

    fun delete(staff: Staff)

    fun deleteByPerformanceId(performanceId: Long?)
}
